Build error with clang-19

**Describe your environment** 

Build HEAD with clang-19

**Steps to reproduce**

Using the environment having clang-19 such as the docker image, silkeh/clang:19, build opentelemetry-cpp with cmake.

**What is the expected behavior?**

Built succesfully.

**What is the actual behavior?**

Clang 19 throws compilation errors as follows

```
opentelemetry-cpp/api/include/opentelemetry/logs/logger_type_traits.h:164:69: error: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
  164 |     return LogRecordSetterTrait::template Set(
      |                                                                     ^

opentelemetry-cpp/api/include/opentelemetry/logs/logger.h:76:89: error: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
   76 |         detail::LogRecordSetterTrait::type>::template Set(
      |                                                                                         ^

opentelemetry-cpp/api/include/opentelemetry/logs/event_logger.h:73:89: error: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
   73 |         detail::LogRecordSetterTrait::type>::template Set(
      |                                                                                         ^
```

**Additional context**

Clang-18 is able to build it so this is a new issue from clang-19. https://github.com/llvm/llvm-project/issues/94194 is the reference point about this issue. Note that `missing-template-arg-list-after-template-kw` is treated as an error by default.

--- a/api/include/opentelemetry/logs/event_logger.h
+++ b/api/include/opentelemetry/logs/event_logger.h
@@ -65,9 +65,8 @@ public:
     }
     nostd::unique_ptr<LogRecord> log_record = delegate_logger->CreateLogRecord();
 
-    IgnoreTraitResult(
-        detail::LogRecordSetterTrait<typename std::decay<ArgumentType>::type>::template Set(
-            log_record.get(), std::forward<ArgumentType>(args))...);
+    IgnoreTraitResult(detail::LogRecordSetterTrait<typename std::decay<ArgumentType>::type>::Set(
+        log_record.get(), std::forward<ArgumentType>(args))...);
 
     EmitEvent(event_name, std::move(log_record));
   }

--- a/api/include/opentelemetry/logs/logger.h
+++ b/api/include/opentelemetry/logs/logger.h
@@ -72,9 +72,8 @@ public:
       return;
     }
 
-    IgnoreTraitResult(
-        detail::LogRecordSetterTrait<typename std::decay<ArgumentType>::type>::template Set(
-            log_record.get(), std::forward<ArgumentType>(args))...);
+    IgnoreTraitResult(detail::LogRecordSetterTrait<typename std::decay<ArgumentType>::type>::Set(
+        log_record.get(), std::forward<ArgumentType>(args))...);
 
     EmitLogRecord(std::move(log_record));
   }

--- a/api/include/opentelemetry/logs/logger_type_traits.h
+++ b/api/include/opentelemetry/logs/logger_type_traits.h
@@ -166,8 +166,8 @@ struct LogRecordSetterTrait
                 * = nullptr>
   inline static LogRecord *Set(LogRecord *log_record, ArgumentType &&arg) noexcept
   {
-    return LogRecordSetterTrait<common::KeyValueIterable>::template Set(
-        log_record, std::forward<ArgumentType>(arg));
+    return LogRecordSetterTrait<common::KeyValueIterable>::Set(log_record,
+                                                               std::forward<ArgumentType>(arg));
   }
 
   template <class ArgumentType,

